ntp-proto/src/nts_record.rs

```diff
@@ -1027,6 +1315,31 @@ mod test {
         assert!(decoder.step().unwrap().is_none());
     }
 
+    #[test]
+    fn server_decoder_finds_algorithm() {
+        let mut bytes = Vec::with_capacity(1024);
+        for record in NtsRecord::client_key_exchange_records() {
+            record.write(&mut bytes).unwrap();
+        }
+
+        let mut decoder = KeyExchangeServerDecoder::new();
+
+        let decode_output = || {
+            for chunk in bytes.chunks(24) {
+                decoder = match decoder.step_with_slice(chunk) {
+                    ControlFlow::Continue(d) => d,
+                    ControlFlow::Break(done) => return done,
+                };
+            }
+
+            Err(KeyExchangeError::IncompleteResponse)
+        };
```

Comment:
this can be made a bit nicer with labelled blocks. Needs a newer rust again though